ReverseTicTacToe/UI/ConsoleUI.cs

```diff
@@ -33,17 +29,145 @@ public  void Init()
+                    if(isInRage(digit1,0,m_Game.BoardSize) &&
+                        (isInRage(digit2,0,m_Game.BoardSize)))
+                    {
+                        turnInputIsValid = v_IsValid;
+                        i_Row = digit1;
+                        i_Col = digit2;
+                    }
+                }
+            }
+            return turnInputIsValid;
+        }
+
+        private bool isInRage(int i_Num, int i_Min, int i_Max)
+        {
+            return i_Num >= i_Min && i_Num <= i_Max;
+        }
+
+        public string GetTurnMessage()
+        {
+            string TurnMsg;
+            if((Game.eTurns)m_Game.CurrentTurn == Game.eTurns.PLAYER1)
+            {
+                TurnMsg = "Player 1 your turn, please enter where to place your symbol (row, col): "
+                          + Environment.NewLine;
+            }
+            else
+            {
+                if(m_Game.GameType == Game.eGameTypes.PVP)
+                {
+                    TurnMsg = "Player 2 your turn, please enter where to place your symbol (row, col): "
+                              + Environment.NewLine;
+                }
+                else
+                {
+                    TurnMsg = "The computer is now calculating what's the best next move is:"
+                              + Environment.NewLine;
+                }
+            }
+            return TurnMsg;
+        }
+
+        private void printGameBoard()
+        {
+            StringBuilder boardOutPut = new StringBuilder();
+            int boardLength = m_Game.BoardSize;
+            for (int i = 0; i <= boardLength; i++)
             {
+                for(int j = 0; j <= boardLength; j++)
+                {
+                    if(i == 0 && j == 0)
+                    {
+                        boardOutPut.Append("  ");
+                    }
+                    else if(i == 0)
+                    {
+                        boardOutPut.Append(j);
+                        boardOutPut.Append("   ");
+                    }
+                    else if(j == 0)
+                    {
+                        boardOutPut.Append(i);
+                    }
+                    else
+                    {
+                    boardOutPut.AppendFormat("| {0} ", m_Game.GameBoard[i - 1, j - 1]);
+                    }
+                }
+
+                if(i > 0)
+                {
+                    boardOutPut.Append("|");
+                }
 
+                boardOutPut.Append(Environment.NewLine);
+                boardOutPut.Append(' ');
+                boardOutPut.Append('=', (boardLength * 4) + 1);
+                boardOutPut.Append(Environment.NewLine);
             }
+            WriteLine(boardOutPut);
         }
 
-        private void getBoardSizeFromUser()
+        private int getBoardSizeFromUser()
         {
             string userInputStr;
             StringBuilder msg = new StringBuilder();
```

Comment:
i think we can make it more reusable here by generelazing the range -  instead of 3 - 9, use constants

Reply:
Good idea